osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/InstallationServiceTest.java

```diff
@@ -144,6 +147,92 @@ void addMeterNoDeviceIdentification() {
     deviceDto.setMbusDefaultKey(new byte[16]);
     // WHEN
     Assertions.assertThatExceptionOfType(FunctionalException.class)
-        .isThrownBy(() -> this.testService.addMeter(this.messageMetadata, deviceDto));
+        .isThrownBy(() -> this.installationService.addMeter(this.messageMetadata, deviceDto));
+  }
+
+  @Test
+  void testAddMeterWithNoExistingDeviceIsNullAndOverwriteIsTrue() throws FunctionalException {
+    final SmartMeteringDeviceDto deviceDto = getSmartMeteringDeviceDto(true);
+    final DlmsDevice dlmsDevice = new DlmsDevice();
+
+    when(this.installationMapper.map(deviceDto, DlmsDevice.class)).thenReturn(dlmsDevice);
+    when(this.dlmsDeviceRepository.findByDeviceIdentification(deviceDto.getDeviceIdentification()))
+        .thenReturn(null);
+
+    final InstallationService installationServiceSpy = Mockito.spy(this.installationService);
+
+    doNothing()
+        .when(installationServiceSpy)
+        .storeAndActivateKeys(any(MessageMetadata.class), any(SmartMeteringDeviceDto.class));
+
+    installationServiceSpy.addMeter(
+        MessageMetadata.newBuilder().withCorrelationUid("123456").build(), deviceDto);
+
+    verify(this.dlmsDeviceRepository, times(1)).save(dlmsDevice);
+  }
+
+  @Test
+  void testAddMeterWithNoExistingDeviceIsNullAndOverwriteIsFalse() throws FunctionalException {
```

Comment:
This test only differs from the previous (testAddMeterWithNoExistingDeviceIsNullAndOverwriteIsTrue) except for the overwrite boolean. Please use a parameterized test for this.

Reply:
Right, tests are updated.